lib/aytests/image_builder.rb

```diff
@@ -92,7 +93,7 @@ def upgrade(autoinst, iso_url)
       setup_definition(:upgrade)
       change_boot_order
       backup_image
-      build
+      build(build_environment(autoinst_path.sub_ext(".linuxrc")))
```

Comment:
It's part of my 'semi-functional' mindset :) For example, it makes the method easier to test. But I can change it anyway.